From 06120f600f2c214fb5f90c9ddfd5e5bc38c0731f Mon Sep 17 00:00:00 2001
From: Martin Schwan <m.schwan@phytec.de>
Date: Thu, 15 Oct 2020 13:35:35 +0200
Subject: [PATCH 1/2] src/update_handler: Reorder *.img.caibx fallback handler

Move the fallback handler for *.img.caibx files to the bottom, so other
default handlers (such as "boot-*") do not get omitted.

Signed-off-by: Martin Schwan <m.schwan@phytec.de>
---
 src/update_handler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/update_handler.c b/src/update_handler.c
index 9265efe..9ddc3f9 100644
--- a/src/update_handler.c
+++ b/src/update_handler.c
@@ -1714,7 +1714,6 @@ RaucUpdatePair updatepairs[] = {
 	{"*.img.caibx", "ubivol", img_to_ubivol_handler},
 	{"*.img.caibx", "ubifs", img_to_ubifs_handler},
 	{"*.squashfs.caibx", "ubivol", img_to_ubivol_handler},
-	{"*.img.caibx", "*", img_to_raw_handler}, /* fallback */
 	{"*.caidx", "ext4", archive_to_ext4_handler},
 	{"*.caidx", "ubifs", archive_to_ubifs_handler},
 	{"*.caidx", "vfat", archive_to_vfat_handler},
@@ -1748,6 +1747,7 @@ RaucUpdatePair updatepairs[] = {
 	{"*.img", "boot-gpt-switch", img_to_boot_gpt_switch_handler},
 #endif
 	{"*", "boot-gpt-switch", NULL},
+	{"*.img.caibx", "*", img_to_raw_handler}, /* fallback */
 	{"*.img", "*", img_to_raw_handler}, /* fallback */
 	{0}
 };
-- 
2.23.0

